### PR TITLE
Add ESP-C3-M1-I-Kit board

### PR DESCRIPTION
**Pull request by wmjohnanderson**
**Summary**
PR for ESP-C3-M1-I-Kit board in variants

I am current summer intern learning as much as I can and using this as practice. This is my first time doing this so if there is anything wrong, or any issues let me know and I can work on it. Thanks!

### CODE REVIEW

**Comment by ajsb85** [comment]
Dear @me-no-dev 
For your references:
https://github.com/platformio/platform-espressif32/pull/841

https://docs.ai-thinker.com/en/esp32c3
https://docs.ai-thinker.com/_media/esp32/docs/esp-c3-m1-i_v1.1.1_specification.pdf
https://docs.ai-thinker.com/_media/esp32/docs/nodemcu-esp-c3-m1-i-kit_v1.2.0_specification.pdf

Best regards.

**Comment by ajsb85** [comment]
Hi, @wmjohnanderson 
Thank you for the contribution.
Could you please rename the folder to create a proper slug name.
Example: from ESP-C3-M1-I-Kit is **esp_c3_m1_i_kit**
Thank you in advance.

**Comment by atanisoft** [comment]
This creates a pins variant but doesn't define any board to use it. Please update boards.txt to have a reference to this new board.

File: esp_c3_m1_i_kit/pins_arduino.h [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+static const uint8_t TX = 18;
+static const uint8_t RX = 19;
+
+static const uint8_t SDA = 4;
+static const uint8_t SCL = 5;
+
+static const uint8_t SS    = 23;
+static const uint8_t MOSI  = 21;
+static const uint8_t MISO  = 22;
+static const uint8_t SCK   = 28;
```
Comment (atanisoft) [inline esp_c3_m1_i_kit/pins_arduino.h:31]: the above four pins are not recommended for use or do not exist on the ESP32-C3. These should be remapped to valid pins.

File: esp_c3_m1_i_kit/pins_arduino.h [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+static const uint8_t A0 = 1;
+static const uint8_t A1 = 2;
+static const uint8_t A2 = 4;
+static const uint8_t A3 = 5;
+static const uint8_t A4 = 6;
+static const uint8_t A5 = 13;
```
Comment (atanisoft) [inline esp_c3_m1_i_kit/pins_arduino.h:38]: GPIO 0 - 5 are the only ADC pins, GPIO 13 does not exist.

**Comment by SuGlider** [comment]
@wmjohnanderson @ajsb85 

Please verify the suggestions and review from @atanisoft in order to move forward.
Thanks!

**Comment by VojtechBartoska** [comment]
@wmjohnanderson Any updates so we can move forward with this PR?

**Comment by VojtechBartoska** [comment]
@wmjohnanderson Can you help process suggested changes?

File: esp_c3_m1_i_kit/pins_arduino.h (hunk 2 above)
Comment (SuGlider) [reply esp_c3_m1_i_kit/pins_arduino.h:38]: @wmjohnanderson - suggestion:
```
static const uint8_t A0 = 0;
static const uint8_t A1 = 1;
static const uint8_t A2 = 2;
static const uint8_t A3 = 3;
static const uint8_t A4 = 4;
static const uint8_t A5 = 5;
```

File: esp_c3_m1_i_kit/pins_arduino.h [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+#define EXTERNAL_NUM_INTERRUPTS 30
+#define NUM_DIGITAL_PINS        30
+#define NUM_ANALOG_INPUTS       6
+
+#define analogInputToDigitalPin(p)  (((p)<NUM_ANALOG_INPUTS)?(analogChannelToDigitalPin(p)):-1)
+#define digitalPinToInterrupt(p)    (((p)<NUM_DIGITAL_PINS)?(p):-1)
+#define digitalPinHasPWM(p)         (p < EXTERNAL_NUM_INTERRUPTS)
+
+static const uint8_t LED_BUILTIN = 18;
+static const uint8_t LED_BUILTIN = 19;
```
[the remark this answers is not in the document]
Comment (SuGlider) [reply esp_c3_m1_i_kit/pins_arduino.h:15]: @wmjohnanderson - suggestion:
```
static const uint8_t LED_C = 18;
static const uint8_t LED_W = 19;
static const uint8_t LED_BUILTIN = 19; 
#define BUILTIN_LED  LED_BUILTIN // backward compatibility
```

File: esp_c3_m1_i_kit/pins_arduino.h (hunk 1 above)
Comment (SuGlider) [reply esp_c3_m1_i_kit/pins_arduino.h:31]: @wmjohnanderson - suggestion:
```
static const uint8_t SS    = 5;
static const uint8_t MOSI  = 10;
static const uint8_t MISO  = 6;
static const uint8_t SCK   = 7;
```

File: esp_c3_m1_i_kit/pins_arduino.h [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+#define BUILTIN_LED  LED_BUILTIN // backward compatibility
+
+static const uint8_t LED_GREEN = 4;
+static const uint8_t LED_RED = 3;
+static const uint8_t LED_BLUE = 5;
+
+static const uint8_t TX = 18;
+static const uint8_t RX = 19;
```
Comment (SuGlider) [inline esp_c3_m1_i_kit/pins_arduino.h:23]: This is the correct pins used in this boards. GPIO18 and 19 are LEDs and USB JTAG/CDC.

```
static const uint8_t TX = 21;
static const uint8_t RX = 20;
```

File: esp_c3_m1_i_kit/pins_arduino.h [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+static const uint8_t SDA = 4;
+static const uint8_t SCL = 5;
```
Comment (SuGlider) [inline esp_c3_m1_i_kit/pins_arduino.h:26]: My suggestion based on the boards layout would be:

```
static const uint8_t SDA = 1;
static const uint8_t SCL = 2;
```

**Comment by SuGlider** [comment]
> This creates a pins variant but doesn't define any board to use it. Please update boards.txt to have a reference to this new board.

@wmjohnanderson - As @atanisoft already said, we also need that you make proper changes to boards.txt in order to add this boards to the list.

**Review by SuGlider — CHANGES_REQUESTED** [review]
@wmjohnanderson  - Please check all the pending requested changes.

**Comment by P-R-O-C-H-Y** [comment]
@wmjohnanderson Hi, is this PR still valid as there is no reaction from you for a long time.

File: esp_c3_m1_i_kit/pins_arduino.h (hunk 3 above)
[the remark this answers is not in the document]
Comment (SuGlider) [reply esp_c3_m1_i_kit/pins_arduino.h]: Fixed.

File: esp_c3_m1_i_kit/pins_arduino.h (hunk 4 above)
Comment (SuGlider) [reply esp_c3_m1_i_kit/pins_arduino.h]: Fixed

File: esp_c3_m1_i_kit/pins_arduino.h (hunk 5 above)
Comment (SuGlider) [reply esp_c3_m1_i_kit/pins_arduino.h]: Fixed with standard ESP32-C3 definition.

File: esp_c3_m1_i_kit/pins_arduino.h (hunk 1 above)
Comment (SuGlider) [reply esp_c3_m1_i_kit/pins_arduino.h]: Fixed.

File: esp_c3_m1_i_kit/pins_arduino.h (hunk 2 above)
Comment (SuGlider) [reply esp_c3_m1_i_kit/pins_arduino.h]: Fixed.

**Review by SuGlider — APPROVED** [review]
Fixed all pending issue in order to move forward and get it merged.

**Comment by SuGlider** [comment]
> @wmjohnanderson Any updates so we can move forward with this PR?

@VojtechBartoska @P-R-O-C-H-Y - I have searched this board speficication and fixed all the pinout definition.
Please review it. I think we can go ahead and get it merged.

**Comment by SuGlider** [comment]
@P-R-O-C-H-Y - Please double check it and if it is ready, please merge it. Thanks!

**Comment by SuGlider** [comment]
> This creates a pins variant but doesn't define any board to use it. Please update boards.txt to have a reference to this new board.

This is done now.

**Review by ajsb85 — APPROVED** [review]

**Comment by me-no-dev** [comment]
Specs say that the module comes with 4MB Flash, so I suggest (strongly) that the unnecessary options are removed (flash size, flash mode, flash freq and partitions that would not fit) 4MB 80Mhz QIO should be default and only option

**Comment by me-no-dev** [comment]
Also this module does not differ in any functional way from the default ESP32-C3 Dev Module

**Comment by me-no-dev** [comment]
@ajsb85 @wmjohnanderson have you tested USB connected to IOs 18 and 19? Does it work OK with the LEDs that you have added?

**Comment by SuGlider** [comment]
> @ajsb85 @wmjohnanderson have you tested USB connected to IOs 18 and 19? Does it work OK with the LEDs that you have added?

The board has a CH340. I think it doesn't mind about USB pins.

**Comment by me-no-dev** [comment]
@SuGlider you still have the options for Flash Frequency and Flash Mode, which I find redundant as well. 4MB 80Mhz QIO is all that is needed

**Comment by SuGlider** [comment]
> @SuGlider you still have the options for Flash Frequency and Flash Mode, which I find redundant as well. 4MB 80Mhz QIO is all that is needed

I think it is fine now. Let me know.

**Comment by ajsb85** [comment]
Hi, @me-no-dev

I will take a look all the features and review the PR in deep with that board.

Thank you very much for the merged into the repo.

**Pull request merged**